docs/kernel_finder.rst

```diff
@@ -0,0 +1,183 @@
+.. _kernel_finder:
+
+================
+Kernel Discovery
+================
+
+The primary purpose of the Jupyter Kernel management package is to provide
+a means of discovering kernels that are available for use.  This is accomplished
```

Comment:
'primary purpose' is a bit of a stretch. I've emphasised the discovery & providers part because that's how it differs from jupyter_client, but launching and interacting with kernel instances is equally important.